source/scripting_v3/GTA/World.cs

```diff
@@ -229,6 +229,42 @@ public static float GravityLevel
 			}
 		}
 
+		/// <summary>
+		/// Gets or sets the rain level.
+		/// </summary>
+		/// <remarks>
+		/// <para>
+		/// Level higher than <c>'0.5f'</c>, only the creation of puddles gets faster, rain and rain sound won't increase after that.
+		/// </para>
+		/// <para>
+		/// Level of <c>'0.0f'</c> rain and rain sounds are disabled and there won't be any new puddles.
+		/// </para>
+		/// <para>
+		/// To use the rain level of the current weather, set this to <c>`-1f`</c>.
+		/// </para>
+		/// </remarks>
+		/// <value>
+		/// The rain level.
+		/// </value>
+		public static float RainLevel
+		{
+			get => Function.Call<float>(Hash.GET_RAIN_LEVEL);
+			set
+			{
+				if (value < 0f)
+				{
+					value = -1;
+				}
+
+				if (value > 1f)
+				{
+					value = 1f;
+				}
```

Comment:
We could set a new rain level override as is without clamping IMO. The game may set the current rain level to a value more than 1.0 if `Rain` value is more than 1.0 in weather.xml. Try editing that file and see what happens with `GET_RAIN_LEVEL` if you doubt.

Reply:
Yeah, but the getter and the setter being separate properties.
```suggestion
		public static float RainLevel => Function.Call<float>(Hash.GET_RAIN_LEVEL);	
		public static float RainLevelOverride
		{
			set => Function.Call(Hash.SET_RAIN, value);
		}
```